client/connection/manager.go

```diff
@@ -151,23 +161,31 @@ func (manager *connectionManager) startConnection(consumerID, providerID identit
 		return err
 	}
 
+	manager.mutex.Lock()
 	manager.cleanConnection = func() {
 		log.Info(managerLogPrefix, "Closing active connection")
 		manager.status = statusDisconnecting()
 		openvpnClient.Stop()
 		log.Info(managerLogPrefix, "Openvpn client stop requested")
 	}
+	manager.mutex.Unlock()
 
 	go openvpnClientWaiter(openvpnClient, dialog)
 	go manager.consumeOpenvpnStates(stateChannel, vpnSession.ID)
 	return nil
 }
 
 func (manager *connectionManager) Status() ConnectionStatus {
+	manager.mutex.RLock()
```

Comment:
Is there a point to block simultaneous status checks?

Reply:
This is only read-lock, multiple reads can be done at the same time. 
The point for this lock was a complaint by the data race detector.